src/compiler/scala/tools/nsc/transform/patmat/Logic.scala

```diff
@@ -356,7 +364,19 @@ trait Logic extends Debugging  {
           // when sym is true, what must hold...
           implied  foreach (impliedSym  => addAxiom(Or(Not(sym), impliedSym)))
           // ... and what must not?
-          excluded foreach (excludedSym => addAxiom(Or(Not(sym), Not(excludedSym))))
+          excluded foreach {
+            excludedSym =>
```

Comment:
I missed that this was a search in a `List`, changed it to a `Set` now. How about keeping the two set lookups for performance reasons?